pkg/terminal/command.go

```diff
@@ -783,11 +783,18 @@ func scopePrefixSwitch(t *Term, ctx callContext) error {
 	return nil
 }
 
+func exitedToError(state *api.DebuggerState, err error) (*api.DebuggerState, error) {
```

Comment:
Thinking through this, we already have a specific error type for this in `pkg/proc`, does that get swallowed before making it out to the client? Can we propagate that error through to the client instead of detecting based off of state?

Reply:
After 480fc02d507c96f1ff939254d687e946a2642290 in `service/debugger` almost all calls to Command will convert a ProcessExitedError into a DebuggerState with Exited set to true. Before that we only did that for `continue`.

Reply:
Ah, yes, you're right.

@dlsniper from an API consumer point of view, is it more convenient if delve returns an error in this case, or is it fine on your end to detect the `Exited` state for any calls that manipulate process execution?

Reply:
From a consumer point of view, it's simpler to handle the exit logic of the process regardless of the way it terminated, either by normal means, Exit(0), via an unexpected Exit(2), or via a panic(). I'm sorry the burden has to be on the server in this case.

Reply:
We've always returned `Exited=true` instead of an error for `continue`, we can't change that because it would be a major backwards compatibility violation. 
`next`, `step` and `stepout` used to return an error instead and 480fc02 changed them to match the behavior of `continue`. Maybe we shouldn't have done that, but we did. At this point, I would prefer to keep them consistent with `continue` than to change them back.

Reply:
Sorry, I wasn't very clear. I definitely think that all operations should behave the same, regardless of operation performed, for the exit behavior, which is what that PR attempted to do. Unfortunately I haven't tested it enough and introduced this bug.

Reply:
> Sorry, I wasn't very clear

You were clear, I was agreeing with you.

Reply:
> At this point, I would prefer to keep them consistent with continue than to change them back.

Agreed.